--- a/src/functions.php
+++ b/src/functions.php
@@ -20,7 +20,7 @@ use function Amp\Promise\any;
  * @return callable Callable executing in another thread / process.
  * @throws SerializationException If the passed callable is not safely serializable.
  */
-function parallel(callable $callable, Pool $pool = null): callable
+function parallel(callable $callable, ?Pool $pool = null): callable
 {
     if ($callable instanceof \Closure) {
         $callable = new SerializableClosure($callable);
@@ -51,7 +51,7 @@ function parallel(callable $callable, Pool $pool = null): callable
  * @return Promise Resolves to the result once the operation finished.
  * @throws \Error If the passed callable is not safely serializable.
  */
-function parallelMap(array $array, callable $callable, Pool $pool = null): Promise
+function parallelMap(array $array, callable $callable, ?Pool $pool = null): Promise
 {
     return call(function () use ($array, $callable, $pool) {
         // Amp\Promise\any() guarantees that all operations finished prior to resolving. Amp\Promise\all() doesn't.
@@ -77,7 +77,7 @@ function parallelMap(array $array, callable $callable, Pool $pool = null): Pro
  * @return Promise
  * @throws \Error If the passed callable is not safely serializable.
  */
-function parallelFilter(array $array, callable $callable = null, int $flag = 0, Pool $pool = null): Promise
+function parallelFilter(array $array, ?callable $callable = null, int $flag = 0, ?Pool $pool = null): Promise
 {
     return call(function () use ($array, $callable, $flag, $pool) {
         if ($callable === null) {

