--- a/Lookahead.php
+++ b/Lookahead.php
@@ -93,6 +93,7 @@
      *
      * @return  \Iterator
      */
+    #[ReturnTypeWillChange]
     public function getInnerIterator()
     {
         return $this->_iterator;
@@ -103,6 +104,7 @@
      *
      * @return  mixed
      */
+    #[ReturnTypeWillChange]
     public function current()
     {
         return $this->_current;
@@ -113,6 +115,7 @@
      *
      * @return  mixed
      */
+    #[ReturnTypeWillChange]
     public function key()
     {
         return $this->_key;
@@ -123,6 +126,7 @@
      *
      * @return  void
      */
+    #[ReturnTypeWillChange]
     public function next()
     {
         $innerIterator = $this->getInnerIterator();
@@ -143,6 +147,7 @@
      *
      * @return  void
      */
+    #[ReturnTypeWillChange]
     public function rewind()
     {
         $out = $this->getInnerIterator()->rewind();
@@ -156,6 +161,7 @@
      *
      * @return  bool
      */
+    #[ReturnTypeWillChange]
     public function valid()
     {
         return $this->_valid;
